File: core/src/main/java/glitch/auth/objects/impl/ValidateImpl.java
package glitch.auth.objects.impl;

import glitch.auth.Scope;
import glitch.auth.objects.Validate;
import lombok.Data;

import java.util.Set;

@Data
public class ValidateImpl implements Validate {
    private final String clientId;

    private final String login;

    private final Set<Scope> scopes;

    private final Long userId;
}

File: core/src/main/kotlin/glitch/auth/objects/json/impl/ValidateImpl.kt
package glitch.auth.objects.json.impl

import glitch.auth.GlitchScope
import glitch.auth.objects.json.Validate

data class ValidateImpl(
        override val clientId: String,
        override val login: String,
        override val scopes: Set<GlitchScope>,
        override val userId: Long
) : Validate
